DigraphyApi/Controllers/TodoController.cs

```diff
@@ -51,48 +42,39 @@ public IActionResult GetTodo(int todoId)
     [ProducesResponseType(400)]
     [ProducesResponseType(204)]
     [ProducesResponseType(404)]
-    public IActionResult UpdateTodo(int todoId,
+    public async Task<IActionResult> UpdateTodo(int todoId,
         [FromBody] PutTodoDto updatedTodo)
     {
-        if (updatedTodo == null)
-            return BadRequest(ModelState);
-
-        if (!_todoRepository.TodoExistsAsync(todoId))
+        if (!(await todoRepository.TodoExistsAsync(todoId)))
             return NotFound();
 
         if (!ModelState.IsValid)
             return BadRequest();
         
-        var existingTodo = _todoRepository.GetTodoAsync(todoId);
+        var existingTodo = await todoRepository.GetTodoAsync(todoId);
         
         if (!string.IsNullOrEmpty(updatedTodo.Name))
         {
-            existingTodo.Name = updatedTodo.Name;
+            if (existingTodo != null) existingTodo.Name = updatedTodo.Name;
         }
 
-        if (!_todoRepository.UpdateTodoAsync(existingTodo))
-        {
-            ModelState.AddModelError("", "Something went wrong updating todo");
-            return StatusCode(500, ModelState);
-        }
+        if (existingTodo == null || await todoRepository.UpdateTodoAsync(existingTodo)) return NoContent();
+        ModelState.AddModelError("", "Something went wrong updating todo");
+        return StatusCode(500, ModelState);
```

Comment:
I would actually keep what you had before. I personally feel like it makes the most sense when the success logic is at the bottom and all the fail conditions are before it, as this feels analogous to guard conditions.